tests-api/spec/use_cases/calculate_score_spec.rb

```diff
@@ -128,8 +128,9 @@
     end
     let(:score) { "#{answers.size}/#{answers.size}" }
     let(:results) do
-      test.questions.map { |q| { question_uuid: q.id, correct: true } }
+      test.questions.map { |q| { question_uuid: q.id, correct_answers: expected_answers, correct: true } }
     end
+    let(:expected_answers) { answers.map { |answer| answer[:selected_choices][0] } }
```

Comment:
since the logic is getting complicated, I encourage you to write a tests for when `result` contains correct and incorrect answers
 
```ruby
let(:results) do
  [ { question_uuid: "question_uuid", correct_answers: ["whatever"], correct: true }]  # context with correct submitted answers
end

let(:results) do
  [ { question_uuid: "question_uuid", correct_answers: ["whatever"], correct: false }]  # context with incorrect submitted answers
end